pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -489,10 +489,23 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusterForPortOrUDS(pluginPara
 	cfg := pluginParams.Push.DestinationRule(pluginParams.Node, instance.Service)
 	if cfg != nil {
 		destinationRule := cfg.Spec.(*networking.DestinationRule)
+		var connectionPool *networking.ConnectionPoolSettings
 		if destinationRule.TrafficPolicy != nil {
+			for _, portTrafficPolicy := range destinationRule.TrafficPolicy.PortLevelSettings {
+				if portTrafficPolicy.GetPort().GetNumber() == uint32(instance.ServicePort.Port) {
```

Comment:
Try to reuse `SelectTrafficPolicyComponents`

Reply:
Thanks for the suggestion, indeed reusable